packages/commercetools/theme/pages/Checkout/Shipping.vue

```diff
@@ -211,39 +251,120 @@ export default {
       loadDetails,
       loading
     } = useCheckout();
+    const { addresses: shippingAddresses, load: loadShippingAddresses, setDefault } = useUserShipping();
+    const { isAuthenticated } = useUser();
+
+    const provideAddress = ref(true);
+    const justChangedAddress = ref(false);
+    const currentAddressId = ref(-1);
+    const setAsDefault = ref(false);
+
+    const findAddressById = id => shippingAddresses.value.find(address => address.id === id);
+
+    const setCurrentAddress = async (addressId) => {
+      const chosenAddress = findAddressById(addressId);
+      if (!chosenAddress) {
+        return;
+      }
+      currentAddressId.value = addressId;
+      setShippingDetails({
+        ...shippingDetails.value,
+        contactInfo: {
+          ...shippingDetails.value.contactInfo,
+          phone: chosenAddress.phoneNumber
+        },
+        streetNumber: chosenAddress.apartment,
+        city: chosenAddress.city,
+        country: chosenAddress.country,
+        state: chosenAddress.state,
+        firstName: chosenAddress.firstName,
+        lastName: chosenAddress.lastName,
+        streetName: chosenAddress.streetName,
+        postalCode: chosenAddress.zipCode
+      });
+      justChangedAddress.value = true;
+    };
+
     onSSR(async () => {
       await loadDetails();
       await loadShippingMethods();
     });
+
+    onMounted(async () => {
+      if (isAuthenticated.value) {
```

Comment:
Soon I will block access to user profile pages for non-logged in users, so there is no need to do it here.

Reply:
But it is checkout's view.